lib/pluck_each.rb

```diff
@@ -28,13 +31,22 @@ def pluck_in_batches(*column_names)
       batch_relation = relation
 
       loop do
-        batch = batch_relation.pluck(*column_names)
-        ids = batch.map(&:first)
+        batch = batch_relation.pluck(*string_column_names)
+        ids = batch.map { |record| record.at(id_position_in_response) }
 
         break if ids.empty?
 
         primary_key_offset = ids.last
 
+        unless id_in_columns_requested
+          batch.collect! do |record|
+            record.delete_at(id_position_in_response)
+            record
+          end
+        end
+
+        batch.flatten! if column_names.size == 1
```

Comment:
it won't do that though .... we append "id" to the pluck call in the instance where it isn't already present so we can use it ... therefore the pluck call will always return an array (unless only id is requested)

we could capture and optimize that path, but doesn't feel worth it to me